### PR TITLE
[MAINTENANCE] Enable `SIM101` + `SIM114` 

### PR DESCRIPTION
**Pull request by Kilo59**
Part 1 of multiple PR's to enable the `SIM` rules.
https://docs.astral.sh/ruff/rules/#flake8-simplify-sim

### Rules included
* SIM101- https://docs.astral.sh/ruff/rules/duplicate-isinstance-call/
* SIM114 - https://docs.astral.sh/ruff/rules/if-with-same-arms/

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -328,6 +330,26 @@ lint.ignore = [
...
+    "SIM105", # suppressible-exception
+    "SIM117", # multiple-with-statements
+    "SIM201", # negate-equal-op
+    "SIM110", # reimplemented-builtin 
+    "SIM910", # dict-get-with-none-default
+    "SIM401", # if-else-block-instead-of-dict-get
+    "SIM115", # open-file-with-context-handler
+    "SIM202", # negate-not-equal-op
+    "SIM210", # if-expr-with-true-false
+    "SIM112", # uncapitalized-environment-variables
```
Comment (Kilo59) [inline pyproject.toml]: This ones still need to be enabled in followup PRs

File: great_expectations/data_context/store/tuple_store_backend.py [hunk 2]
```diff
@@ -76,6 +76,14 @@ def __init__(  # noqa: PLR0913
             self.verify_that_key_to_filepath_operation_is_reversible()
             self._fixed_length_key = True
 
+    @staticmethod
```
Comment (anthonyburdi) [inline great_expectations/data_context/store/tuple_store_backend.py:79]: I'm ok if we remove the `@staticmethod` decorator or leave it. I don't have a strong argument either way.

File: great_expectations/datasource/fluent/snowflake_datasource.py [hunk 3]
```diff
@@ -136,12 +136,16 @@ def _check_xor_input_args(cls, values: dict) -> dict:
...
+            )
             # Method 2 - individual args (account, user, and password are bare minimum)
-            elif isinstance(connection_string, ConnectionDetails) and bool(
+            has_min_connection_detail_values: bool = isinstance(
+                connection_string, ConnectionDetails
+            ) and bool(
                 connection_string.account and connection_string.user and connection_string.password
-            ):
+            )
+            if is_connection_string or has_min_connection_detail_values:
```
Comment (anthonyburdi) [inline great_expectations/datasource/fluent/snowflake_datasource.py:148]: Nice!

**Review by anthonyburdi — APPROVED** [review]
LGTM! 
I think `tests/test_definitions/test_expectations_v3_api.py` could use some refactoring but I'm OK if that is not included here.

File: great_expectations/data_context/store/tuple_store_backend.py (hunk 2 above)
Comment (Kilo59) [reply great_expectations/data_context/store/tuple_store_backend.py:79]: If it isn't relying on instance or class attributes I think it's better as a static method.

**Pull request merged**